### ISSUE TITLE
Invisible character shown on terminal buffer

### ISSUE THREAD

**Issue by ajiyakin** [neovim/neovim#4392]
- Neovim version: NVIM 0.1.3-dev
- Operating system: Linux
- Terminal emulator: xterm-256color
### Actual behaviour

When I open terminal emulator, hidden characters become visible since I set it on my `init.vim` file using `set list` and `set listchars` and it's pretty odd, especially when I set `nbsp` character within `listchars`.
### Expected behaviour

It would be nice if the terminal only show the visible characters only by default, unless user specified it manually.

Here is when I set `nbsp` character on my `init.vim` file:
![screenshot from 2016-03-02 21 52 20](https://cloud.githubusercontent.com/assets/7039123/13463930/297ed166-e0c1-11e5-9fd1-7b5469a5c440.png)

---

I don't know if this issue has been reported or if I miss something please tell me, I just can't find it on issue page.
If this is the default behavior on terminal buffer, how I can configure it separately with another buffers.

**Comment by justinmk** [comment]
Problem is that `'list'` is window-local. So there is no standard mechanism for restoring the option when a non-terminal buffer is shown (the logic also isn't clear).

I have been thinking that making `:terminal` act like `:help` (i.e., always open a split) would have several benefits:

- avoids having to manage window-local options
- enables `:tab te {cmd}` to work
- allows us to use the bang `!` for something else

**Comment by ajiyakin** [comment]
That's cool idea to always open a split window @justinmk . I hope it will included in the next release :smile: While I still learn Lua and VimL scripting language.
And after I read the manual, there are event handler called `TermOpen` that can handle my expectation above. Sorry, my bad to know it lately. I finally add this configuration to my `init.vim` file:

```
:autocmd TermOpen * if &buftype == 'terminal' | :set nolist | endif
```

And it works.

**Comment by justinmk** [comment]
Closed by #6246
